src/client.ts

```diff
@@ -138,13 +143,13 @@ export class Client {
    * due to an internal error.
    */
   async query<T = any>(
-    request: QueryRequest | QueryBuilder,
-    headers?: QueryRequestHeaders
+    request: (QueryRequest & QueryRequestOptions) | QueryBuilder,
+    requestOptions?: QueryRequestOptions
   ): Promise<QuerySuccess<T>> {
```

Comment:
Two places this change turned `QueryRequest` into `(QueryRequest & QueryRequestOptions)`. Which is kinda ugly.

> Note, you can embed request options in this object; if you do that there's no need to pass the requestOptions parameter.

I would like to argue that we shouldn't overload the request parameter like this. But right now this PR doesn't change any functionality, whereas dropping `& QueryRequestOptions` would break the contract we've had for a long time.

Reply:
@ptpaterson its part of the wire protocol that any header can be sent as a request field.

That's why it was done this way.

Reply:
well hell

Reply:
Actually... not anymore? I don't see that in the openAPI spec.

Reply:
Confirmed by core team that we've removed the options from the query body and only accept through headers.